src/Form/tests/FormAutosuggest.test.jsx

```diff
@@ -25,141 +14,179 @@ function FormAutosuggestWrapper(props) {
       floatingLabel="floatingLabel text"
       helpMessage="Example help message"
       errorMessageText="Example error message"
-      onSelected={onSelected}
+      onSelected={props.onSelected}
     >
       <FormAutosuggestOption>Option 1</FormAutosuggestOption>
-      <FormAutosuggestOption onClick={onClick}>Option 2</FormAutosuggestOption>
+      <FormAutosuggestOption onClick={props.onClick}>Option 2</FormAutosuggestOption>
       <FormAutosuggestOption>Learn from more than 160 member universities</FormAutosuggestOption>
-    </FormAutosuggestWrapper>,
+    </FormAutosuggestWrapper>
   );
+}
 
-  describe('render behavior', () => {
-    it('renders component without error', () => {
-      mount(<FormAutosuggestWrapper />);
-    });
+FormAutosuggestTestComponent.defaultProps = {
+  onSelected: jest.fn(),
+  onClick: jest.fn(),
+};
 
-    it('render without loading state', () => {
-      expect(container.exists('.pgn__form-autosuggest__dropdown-loading')).toBe(false);
-      expect(container.props().isLoading).toBeUndefined();
-    });
+FormAutosuggestTestComponent.propTypes = {
+  /** Specifies onSelected event handler. */
+  onSelected: PropTypes.func,
+  /** Specifies onClick event handler. */
+  onClick: PropTypes.func,
+};
 
-    it('render with loading state', () => {
-      const wrapper = mount(<FormAutosuggestWrapper isLoading />);
+describe('render behavior', () => {
+  it('renders component without error', () => {
+    render(<FormAutosuggestWrapper />);
+  });
 
-      expect(wrapper.exists('.pgn__form-autosuggest__dropdown-loading')).toBe(true);
-      expect(wrapper.props().isLoading).toBe(true);
-    });
+  it('renders without loading state', () => {
+    const { container } = render(<FormAutosuggestTestComponent />);
+    expect(container.querySelector('.pgn__form-autosuggest__dropdown-loading')).toBeNull();
+  });
 
-    it('renders the auto-populated value if it exists', () => {
-      const wrapper = mount(<FormAutosuggestWrapper value="Test Value" />);
+  it('render with loading state', () => {
+    const { container } = render(<FormAutosuggestWrapper isLoading />);
+    expect(container.querySelector('.pgn__form-autosuggest__dropdown-loading')).toBeTruthy();
+  });
 
-      expect(wrapper.find('input').instance().value).toEqual('Test Value');
-      expect(wrapper.props().value).toEqual('Test Value');
-    });
+  it('renders the auto-populated value if it exists', () => {
+    render(<FormAutosuggestWrapper value="Test Value" />);
+    expect(screen.getByDisplayValue('Test Value')).toBeInTheDocument();
+  });
 
-    it('renders component with options', () => {
-      container.find('input').simulate('click');
-      const optionsList = container.find('.pgn__form-autosuggest__dropdown').find('li');
+  it('renders component with options', () => {
+    const { getByTestId, container } = render(<FormAutosuggestTestComponent />);
+    const input = getByTestId('autosuggest_textbox_input');
+    fireEvent.click(input);
```

Comment:
[curious] Is there a reason to use `fireEvent` here (and elsewhere throughout the tests) versus using `userEvent`? 

If it's possible to use `userEvent.click` instead, it's generally preferable since it simulates more of the true user interactions in the browser, not just the low-level click itself. 

For example, `userEvent.click` also simulates a previous `hover` event that a real user would trigger prior to clicking.

Resources:
* https://testing-library.com/docs/guide-events/
* https://blog.mimacom.com/react-testing-library-fireevent-vs-userevent/